### DISCUSSION TITLE
Search for Checkbox with State / Display Checkbox State in Tasklist

### DISCUSSION THREAD

**Discussion by VpGitHub231** [zim-desktop-wiki/zim-desktop-wiki#2151] [Q&A] [answered]
Hi,

as this is my first Question here i want to introduce myself shortly:
I'm using ZIM for some years now, but since last year i moved all my notes, files, etc. to ZIM and now also started to use ZIM for Task management. So that i have everything together in one application.
And i really like ZIM with it's unique functionality which perfectly fits my thinking/mindset/usecase.
At this point a big Thanks to all contributors for your work!

Now my question:

As mentioned i use the Task List Plugin and i would need some additional functionality. My usecase: I use the Checkboxes as a sort of „Kanban“ System, were i use the Checkbox State as in indicator of the Task State. For example:

Checkbox [ ] = Task Open
Checkbox [<] = Task Waiting for input
Checkbox [>] = Task Work in progress
Checkbox [*] = Task Done

These Checkboxes are all over the Notebook, for different Projects, Shopping, Goals, etc. and i use the „Turn page name into tags for task items“ option for grouping the Tasks in in the Task List. This works fine and is exactly what i need, but what I'm missing is the following.
The Task List shows only Tasks with Checkbox State OPEN [ ]. I can not see the Tasks with [<] and [>] Checkboxes in the Task List, which would be important in my usecase to see them, as they are also not done yet.

• Maybe it can be implemented in the Task List Plugin to also show Tasks with Checkbox State other than [ ]? I think there could be an option in the Notebook properties to select which Checkbox States with corresponding Task should be shown in the Task List. Then in the Task List itself the Checkbox State should be shown with each Task and also be sortable by the Checkbox State.

• As workaround, is there a way the use the „normal” search function to search for Checkboxes with a defined State? I already read the documentation and could not find something about this topic.

I'm just wondering that nobody else had the need for such a functionality before. I have read all the Issues/Pull-Requests/Discussion on GitHub and the Wiki, but something like this was never mentioned.
But i think this functionality could be useful for a lot of people.

Also a few days ago i have seen some discussion about the new Task List Plugin in the development version, but i can not find it anymore. I'm not sure if there is such a functionality maybe already implemented. If so, then please forget about my post here.
If not, then maybe there is still the possibility to implement something like this before the next release?

Thanks for your efforts and reading this Post, I'm looking forward to your answers.
Vince

**Comment by introt** [comment, marked as answer]
Hi! The new Task List Plugin lets you choose which states are included in the list via the tick boxes in the task window's hamburger menu. There's also a state column that can be sorted by. Feel free to take the development branch for a spin - our Python code needs no complicated compilation :)

**Reply by VpGitHub231** [reply]
Hi, thanks very much for your answer!
Sounds exactly like what i wanted. I will try it out
